tests/jest/components/AdminSettings.spec.js

```diff
@@ -727,6 +793,674 @@ describe('AdminSettings.vue', () => {
+					await wrapper.setData({
+						opUserAppPassword: false,
+					})
+					const radioWitchButton = wrapper.find(selectors.groupFolderSetupSwitch)
+					await radioWitchButton.trigger('click')
+					await wrapper.vm.$nextTick()
+					expect(wrapper.vm.isProjectFolderSwitchEnabled).toBe(true)
+					const setupGroupFolderButton = wrapper.find(selectors.completeGroupFolderSetupWithGroupFolderButton)
+					expect(setupGroupFolderButton.text()).toBe('Setup OpenProject user, group and folder')
+				})
+			})
+
+			describe('upon complete setup (group folder and app password)', function() {
+				describe('edit mode again', function() {
+					let wrapper = {}
+					let getgroupfolderStatusSpy
+					beforeEach(async () => {
+						axios.put.mockReset()
+						axios.get.mockReset()
+						wrapper = getMountedWrapper({
+							state: {
+								openproject_instance_url: 'http://openproject.com',
+								openproject_client_id: 'some-client-id-here',
+								openproject_client_secret: 'some-client-secret-here',
+								nc_oauth_client: {
+									nextcloud_client_id: 'some-nc-client-id-here',
+									nextcloud_client_secret: 'some-nc-client-secret-here',
+								},
+								fresh_group_folder_setup: false,
+								group_folder_status: {
+									status: true,
+								},
+								app_password_set: true,
+							},
+							isGroupFolderAlreadySetup: null,
+						})
+						await wrapper.setData({
+							formMode: {
+								groupFolderSetUp: F_MODES.EDIT,
+							},
+							oPUserAppPassword: 'opUserPassword',
+						})
+						getgroupfolderStatusSpy = jest.spyOn(axios, 'get').mockImplementationOnce(() => Promise.resolve({
+							data: {
+								result: true,
+							},
+						}))
+					})
+
+					it('should show button label to "keep current"', async () => {
+						const setupGroupFolderButton = wrapper.find(selectors.completeGroupFolderSetupWithGroupFolderButton)
+						expect(setupGroupFolderButton.text()).toBe('Keep current change')
+					})
+
+					it('on trigger "Keep on current change" should not create new user app password', async () => {
+						const setupGroupFolderButton = wrapper.find(selectors.completeGroupFolderSetupWithGroupFolderButton)
+						expect(setupGroupFolderButton.text()).toBe('Keep current change')
+						setupGroupFolderButton.trigger('click')
+						expect(getgroupfolderStatusSpy).toBeCalledTimes(1)
+						expect(wrapper.vm.oPUserAppPassword).toBe('opUserPassword')
+					})
+
+					it('on switch "off" should show button label as "Complete without group folder setup"', async () => {
+						const radioWitchButton = wrapper.find(selectors.groupFolderSetupSwitch)
+						await radioWitchButton.trigger('click')
+						const setupGroupFolderButton = wrapper.find(selectors.completeWithoutGroupFolderSetupButton)
+						expect(setupGroupFolderButton.text()).toBe('Complete without group folder setup')
+					})
+
+					it('resetting should set switch as "off" again (same as fresh set up)', async () => {
```

Comment:
Should this be "on" ?
```suggestion
					it('should set switch as "on" after a full reset', async () => {
```